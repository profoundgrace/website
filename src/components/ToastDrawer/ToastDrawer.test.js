import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import ConnectedToastDrawer, { ToastDrawer } from './ToastDrawer';

describe('<ToastDrawer />', () => {
  const queue = [
    {
      id: 'test',
      title: 'I am a toast',
      message: 'Ask me anything'
    }
  ];

  it('renders correctly', () => {
    const props = { toasts: queue };
    const component = renderer.create(<ToastDrawer {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders null if empty', () => {
    const props = { toasts: [] };
    const component = renderer.create(<ToastDrawer {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders connected component correctly', () => {
    const initialState = { toast: { queue } };
    const mockStore = configureStore();
    const store = mockStore(initialState);
    const component = renderer.create(
      <Provider store={store}>
        <ConnectedToastDrawer />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
