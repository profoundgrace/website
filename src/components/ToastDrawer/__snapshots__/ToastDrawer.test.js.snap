// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToastDrawer /> renders connected component correctly 1`] = `
<section
  className="toast-drawer"
>
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
  >
    <div
      className="toast-header"
    >
      <strong
        className="mr-auto"
      >
        I am a toast
      </strong>
      <button
        className="close ml-2 mb-1"
        data-dismiss="toast"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
        <span
          className="sr-only"
        >
          Close
        </span>
      </button>
    </div>
    <div
      className="toast-body"
    >
      Ask me anything
    </div>
  </div>
</section>
`;

exports[`<ToastDrawer /> renders correctly 1`] = `
<section
  className="toast-drawer"
>
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
  >
    <div
      className="toast-header"
    >
      <strong
        className="mr-auto"
      >
        I am a toast
      </strong>
      <button
        className="close ml-2 mb-1"
        data-dismiss="toast"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
        <span
          className="sr-only"
        >
          Close
        </span>
      </button>
    </div>
    <div
      className="toast-body"
    >
      Ask me anything
    </div>
  </div>
</section>
`;

exports[`<ToastDrawer /> renders null if empty 1`] = `null`;
